### PR TITLE
FESBAL-52  Allow relative child without doc id

### PR DESCRIPTION
**Pull request by santigamo**
## Description

When children (under 18 years) are added as family members, adding the document number is not mandatory.

In the backend, we must check the date of birth (which is mandatory) and make the document and document type field mandatory or optional depending on the age.

## Changes

- Changes command, event and entity to allow relatives without doc. id.
- Added age validation in CreateRelative command to throw errors if the relative are older than 18 yo and doesn't have doc. id.

## Checks

- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

### CODE REVIEW

File: src/commands/relative/create-relative.ts [hunk 1]
```diff
@@ -33,4 +37,7 @@ export class CreateRelative {
     register.events(new RelativeAddedToRecipientUser(command.recipientUserId, command.relativeId))
     await register.flush()
   }
+
+  private static getAge = (birthDate: string): number =>
```
Comment (samueldominguez) [inline src/commands/relative/create-relative.ts:41]: This fails when using the same day, also by using time duration in a year we don't take into account leap years. We can use this instead:
```
const getAge = (birthDate: string): number => {
  const now = new Date()
  const birth = new Date(birthDate)
  let age = now.getFullYear() - birth.getFullYear()
  if (now.getMonth() < birth.getMonth() || now.getDate() < birth.getDate()) {
    age--
  }
  return age;
}
```

**Review by samueldominguez — CHANGES_REQUESTED** [review]
Left a comment, everything else LGTM

File: src/commands/relative/create-relative.ts (hunk 1 above)
Comment (samueldominguez) [reply src/commands/relative/create-relative.ts:41]: By same day I mean, test with `2005-04-18` it will output 17.

Comment (santigamo) [reply src/commands/relative/create-relative.ts]: Updated!

**Review by samueldominguez — APPROVED** [review]

**Pull request merged**